cmd/webhook.go

```diff
@@ -0,0 +1,122 @@
+func createWebhookCommand() *cobra.Command {
+	var webhookPort int
+	var certDir, certFile, keyFile string
+	var metricsBindAddress string
+
+	cmd := &cobra.Command{
+		Use:   "resend-webhook",
+		Short: "Runs the Resend webhook server",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runWebhook(
+				cmd,
+				webhookPort,
+				certDir, certFile, keyFile,
+				metricsBindAddress)
+		},
+	}
+
+	// Network & Kubernetes flags.
+	cmd.Flags().IntVar(&webhookPort, "webhook-port", 9443, "Port for the webhook server")
+	cmd.Flags().StringVar(&certDir, "cert-dir", "/etc/certs",
+		"Directory that contains the TLS certs to use for serving the webhook")
+	cmd.Flags().StringVar(&certFile, "cert-file", "", "Filename in the directory that contains the TLS cert")
+	cmd.Flags().StringVar(&keyFile, "key-file", "", "Filename in the directory that contains the TLS private key")
+
+	// Metrics flags.
+	cmd.Flags().StringVar(&metricsBindAddress, "metrics-bind-address", ":8080", "address the metrics endpoint binds to")
+
+	return cmd
+}
+
+func runWebhook(
+	cmd *cobra.Command,
+	webhookPort int,
+	certDir, certFile, keyFile string,
+	metricsBindAddress string) error {
+	logf.SetLogger(zap.New(zap.JSONEncoder()))
+	log := logf.Log.WithName("resend-webhook")
+
+	// Validate command flags early so we fail fast with meaningful errors.
+	if _, err := config.NewWebhookConfig(webhookPort, certDir, certFile, keyFile, metricsBindAddress); err != nil {
+		return err
+	}
+
+	// Setup Kubernetes client config
+	restConfig, err := k8sconfig.GetConfig()
+	if err != nil {
+		return fmt.Errorf("failed to get rest config: %w", err)
+	}
+
+	runtimeScheme := runtime.NewScheme()
+	if err := notificationmiloapiscomv1alpha1.AddToScheme(runtimeScheme); err != nil {
+		return fmt.Errorf("failed to add notificationmiloapiscomv1alpha1 scheme: %w", err)
+	}
+	if err := eventsv1.AddToScheme(runtimeScheme); err != nil {
+		return fmt.Errorf("failed to add eventsv1 scheme: %w", err)
+	}
+
+	log.Info("Creating manager")
+	mgr, err := manager.New(restConfig, manager.Options{
+		Scheme: runtimeScheme,
+		Metrics: server.Options{
+			BindAddress: metricsBindAddress,
+		},
+		WebhookServer: ctrlwebhook.NewServer(ctrlwebhook.Options{
+			CertDir:  certDir,
+			CertName: certFile,
+			KeyName:  keyFile,
+			Port:     webhookPort,
+		}),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create manager: %w", err)
+	}
+
+	log.Info("Adding providerID index")
+	if err := mgr.GetFieldIndexer().
+		IndexField(
+			cmd.Context(),
+			&notificationmiloapiscomv1alpha1.Email{},
+			"status.providerID",
+			func(rawObj client.Object) []string {
+				email := rawObj.(*notificationmiloapiscomv1alpha1.Email)
+				if email.Status.ProviderID == "" {
+					return nil
+				}
+				return []string{email.Status.ProviderID}
+			}); err != nil {
+		return fmt.Errorf("failed to create index for providerID: %w", err)
+	}
```

Comment:
I'd move this into the webhook package and have a `SetupWithManager()` function that can be used to register the webhook with the controller manager and setup field indexers. Similar to how controllers are setup.